indico_jacow/client/PeerReviewManagersFileInput.jsx

```diff
@@ -0,0 +1,215 @@
+// the LICENSE file for more details.
+
+import uploadManagersFileURL from 'indico-url:plugin_jacow.peer_review_managers_import'
+
+import PropTypes from 'prop-types';
+import React, {useCallback, useState} from 'react';
+import {useDropzone} from 'react-dropzone';
+import {Button, Message, MessageHeader, MessageList, 
+    MessageItem, Icon, Loader, Segment, Dimmer} from 'semantic-ui-react';
+import {Field} from 'react-final-form';
+
+import {FinalField, handleSubmitError} from 'indico/react/forms';
+import {Translate} from 'indico/react/i18n';
+
+import {SingleFileArea} from 'indico/react/components/files/FileArea';
+import {FinalModalForm} from 'indico/react/forms/final-form';
+import {indicoAxios} from 'indico/utils/axios';
+
+import './PeerReviewManagerFileInput.module.scss';
+
+const PeerReviewManagersFileInput = ({
+    onFocus,
+    onBlur,
+    onChange,
+    validExtensions,
+    setUnknownEmails,
+    setUserIdentifiers,
+    setLoading,
+    eventId,
+}) => {
+    const [file, setFile] = useState();
+
+    const markTouched = () => {
+        onFocus();
+        onBlur();
+    };
+
+    async function getUserList(file, eventId) {
+        const headers = {'content-type': 'multipart/form-data'}
+        let response;
+        setLoading(true);
+        try {
+            const formData = new FormData();
+            formData.append('file', file);
+            response = await indicoAxios.post(uploadManagersFileURL({event_id: eventId}), formData, {headers});
+            if (response.data.unknown_emails.length > 0) {
+                setUnknownEmails(response.data.unknown_emails)
+            }
+            setUserIdentifiers(response.data.identifiers)
+            return true;
+        } catch (e) {
+            handleSubmitError(e);
+            return false;
+        } finally {
+            setLoading(false);
+        }
+    }
+
+    const onDropAccepted = useCallback(async ([acceptedFile]) => {
+        acceptedFile.filename = acceptedFile.name;
+        const isSuccess = await getUserList(acceptedFile, eventId);
+        if(isSuccess){
+            setFile(acceptedFile);
+            onChange(acceptedFile);
+        }
+    }, [file, setFile, onChange]);
+
+    const dropzone = useDropzone({
+        onDragEnter: markTouched,
+        onFileDialogCancel: markTouched,
+        onDrop: markTouched,
+        onDropAccepted,
+        accept: validExtensions ? validExtensions.map(ext => `.${ext}`) : null,
+        multiple: false,
+        noClick: true,
+        noKeyboard: true,
+    });
+
+    return <SingleFileArea dropzone={dropzone} file={file} />;
+};
+
+
+export default function PeerReviewManagersFileField ({onClose, eventId, onChange}) {
+    const [unknownEmails, setUnknownEmails] = useState([])
+    const [userIdentifiers, setUserIdentifiers] = useState([])
+    const [loading, setLoading] = useState(false)
+
+    const handleSubmit = async () => {
+        onChange(userIdentifiers)
```

Comment:
When you import the same twice you end up with duplicates, and combined with duplicate key that's causing all kinds of issues (try removing the duplicates).

Reply:
I fixed this by overwriting the list. Could generate a merged list as well since we have access to `entries`. Not sure if that's needed though.

Reply:
Looking into this